### PR TITLE
Added abstract base driver class 

### PR DESCRIPTION
**Pull request by kberryUSGS**
Any driver test that doesn't implement everything in this abstract base class will fail. I didn't go through and update all of them in case there were substantial changes requested on the PR. 

This is for #137

### CODE REVIEW

File: ale/base/base.py [hunk 1]
```diff
@@ -61,102 +48,249 @@ def to_dict(self):
...
+    def usgscsm_distortion_model(self):
+        """
+        Returns
+        -------
+        : dict
+          A dict containing the information about the distortion model for the usgscsm
+        """
+        pass
+
+#   def is_valid(self):
```
Comment (kberryUSGS) [inline ale/base/base.py]: Should `is_valid` remain? It wasn't explicitly part of our design, so I commented it out.

File: tests/pytests/test_base.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    def sun_position(self):
+        return ([1], [1])
+
+    @property
+    def target_body_id(self):
+        return 1
+
+def test_me(test_label):
+    me = testclass(test_label)
+    assert me.image_lines == 1
```
Comment (kberryUSGS) [inline tests/pytests/test_base.py]: I'm not sure if the ABC is worked as intended here: `image_lines` is defined in `IsisLabel`, which the test class inherits from, but I had to redefine `image_lines` in the test class in order to not get a 'not implemented' error. Anyone have any ideas about this?

**Comment by jessemapel** [comment]
@scsides has some changes in #154 that may conflict with this, so I'm going to wait until that is merged to start on this.

**Comment by jessemapel** [comment]
COntinued at #163

**Pull request closed without merging**